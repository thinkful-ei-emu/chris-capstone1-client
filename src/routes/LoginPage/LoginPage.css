.LoginPage {
    width: 320px;
    margin-left: auto;
    margin-right: auto;
    background-color: #CC8B56;
    border: 0;
    text-decoration: none;
    box-shadow: 0px 11px 21px -8px rgba(0, 0, 0, 0.5);
    margin-bottom: 40px;
    border-radius: 10px;
    padding: 30px;
}