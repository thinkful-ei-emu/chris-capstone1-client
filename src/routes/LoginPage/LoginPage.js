import React from 'react';
import { Section } from '../../components/Utils/Utils';

export default class LoginPage extends React.Component {
    static defaultProps = {
        location: {},
        history: {
            push: () => {}
        }
    }

    handleLoginSuccess = () => {
        const { location, history } = this.props;
        const destination = (location.state || {}).from || '/home';
        history.push(destination);
    }

    render() {
        return (
            <Section className='LoginPage'>
                <h1>Login</h1>
                <LoginForm
                    onLoginSuccss={this.handleLoginSuccess}
                />
            </Section>
        )
    }
}