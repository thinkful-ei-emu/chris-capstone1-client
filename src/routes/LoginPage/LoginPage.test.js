import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import LoginPage from './LoginPage';
import ReactDom from 'react-dom';
import { BrowserRouter } from 'react-router-dom';

describe('LoginPage component', () => {
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDom.render(
          <BrowserRouter>
              <LoginPage />
          </BrowserRouter>,
          div
        )
        ReactDom.unmountComponentAtNode(div)
      })
    it('renderes LoginPage by default', () => {
        expect(toJson(shallow(<BrowserRouter>
            <LoginPage />
        </BrowserRouter>))).toMatchSnapshot() 
    })
})