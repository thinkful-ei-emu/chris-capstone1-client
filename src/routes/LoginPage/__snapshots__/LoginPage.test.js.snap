// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage component renderes LoginPage by default 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "createHref": [Function],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <LoginPage
    history={
      Object {
        "push": [Function],
      }
    }
    location={Object {}}
  />
</Router>
`;
